### PR TITLE
Test_highlevel: Make test families for highlevel tests

### PR DESCRIPTION
**Pull request by euanh**
With this change, OUnit reports each testcase, rather than
only reporting groups of tests created by Test_highlevel
modules.

Signed-off-by: Euan Harris <euan.harris@citrix.com>

### CODE REVIEW

File: ocaml/test/test_highlevel.ml [hunk 1]
```diff
@@ -88,17 +88,21 @@ module Generic = struct
...
-				~msg:(Printf.sprintf
-					"Failure: input = %s, output = %s, expected output = %s"
-					(T.Io.string_of_input_t input)
-					(T.Io.string_of_output_t actual_output)
-					(T.Io.string_of_output_t expected_output))
-				actual_output expected_output
+			let title = Printf.sprintf "%s -> %s" 
+				(T.Io.string_of_input_t input)
+				(T.Io.string_of_output_t expected_output) in
+			title >:: (fun _ ->
```
Comment (johnelse) [inline ocaml/test/test_highlevel.ml]: Can this `_` be a `()`?

Comment (euanh) [inline ocaml/test/test_highlevel.ml]: Yup.

**Comment by johnelse** [comment]
Looks good. Moar tests :)

**Comment by johnelse** [comment]
'''
- File "test_map_check.ml", line 105, characters 27-43:
- Error: Unbound value AddDefaults.test
- Hint: Did you mean tests?
'''

**Pull request merged**